Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/azure/serverless/AzureSparkServerlessClusterManager.java

```diff
@@ -72,7 +73,8 @@ public static AzureSparkServerlessClusterManager getInstance() {
     private static final String REST_SEGMENT_ADL_ACCOUNT = "providers/Microsoft.DataLakeAnalytics/accounts";
 
     // FIXME!!!
-    private static final String ACCOUNT_FILTER = CommonSettings.getAdEnvironment().endpoints().getOrDefault("dataLakeSparkAccountFilter", "endswith(name, '-c09')");
+    // the last 4 characters match with pattern: -c**
+    private static final String ACCOUNT_FILTER = CommonSettings.getAdEnvironment().endpoints().getOrDefault("dataLakeSparkAccountFilter", "substring(name, length(name) sub 4, 2) eq '-c'");
```

Comment:
I prefer to use filter like `endswith(name, '-c09') or endswith(name, '-c08') or ...` with new clusters introduced.

Reply:
Talked with Rui offline, seems like it will works even length is less than 4.

---
In reply to: [202234728](https://github.com/Microsoft/azure-tools-for-java/pull/1766#discussion_r202234728) [](ancestors = 202234728,202225399)